pkg/operator/controller/ingress_status.go

```diff
@@ -35,26 +36,123 @@ func (r *reconciler) syncIngressControllerStatus(deployment *appsv1.Deployment,
 }
 
 // computeIngressStatusConditions computes the ingress controller's current state.
-func computeIngressStatusConditions(oldConditions []operatorv1.OperatorCondition, deployment *appsv1.Deployment) []operatorv1.OperatorCondition {
+func computeIngressStatusConditions(oldConditions []operatorv1.OperatorCondition, deployment *appsv1.Deployment,
+	uniqueDomain bool) []operatorv1.OperatorCondition {
+	oldDegradedCondition := getIngressDegradedCondition(oldConditions)
+	oldProgressingCondition := getIngressProgressingCondition(oldConditions)
 	oldAvailableCondition := getIngressAvailableCondition(oldConditions)
```

Comment:
I don't know that having separate `getIngressDegradedCondition`, `getIngressProgressingCondition`, and `getIngressAvailableCondition` functions is important for readability, and it means more code and more looping.  What do you think of using a single loop in `computeIngressStatusConditions` to get all three values, similar to how the DNS operator does it? https://github.com/openshift/cluster-dns-operator/blob/540ab8bca50b50880a4eb44feaddee8352f565bf/pkg/operator/controller/status.go#L202-L212

Reply:
@Miciah I was going back and forth on whether to have separate functions for each condition or looping through the conditions in a single function. I thought the former would be easier to read and understand. I will update the PR using a single function.

Reply:
>  Do these changes seem reasonable, and if so, could we incorporate them into this PR, or would it be better to put them in a separate PR?

They do make sense. I'm working on updating the PR.